### ISSUE TITLE
`export let x: Type` and type declaration without ending semicolon last in script block breaks syntax highlighting

### ISSUE THREAD

**Issue by poppa** [sveltejs/language-tools#1970]
### Describe the bug

When running Typescript mode and you have either an `export let varName: Type`, or `type Type = type`  without a line-ending semicolon last in a script block, the syntax highlighting breaks.

The problem with type declarations I've seen for ages, but the problem with component properties arrived yesterday after updating VS Code.

I think the screenshots are quite explanatory.

**Syntax highlight breaks after ZeRow**

<img width="577" alt="export-let" src="https://user-images.githubusercontent.com/92401/229279360-fede1535-26e9-4651-a21e-da22da4accc7.png">

**Syntax highlighting breaks after the ending curly brace for type declaration `Test`**

<img width="552" alt="type" src="https://user-images.githubusercontent.com/92401/229279363-69c9744c-1f46-4d21-8571-105dbd36216b.png">

**Syntax highlight works when ending type declaration with a semicolon**

<img width="578" alt="type-with-semi" src="https://user-images.githubusercontent.com/92401/229279364-b3d430dd-9d08-4eb4-b56f-7a401745ca21.png">

**Syntax highlighting works when ending property declaration with semicolon**

<img width="563" alt="export-let-with-semi" src="https://user-images.githubusercontent.com/92401/229279366-5fdaed57-4821-49af-9ed5-f7a0c0a32a1c.png">

**Syntax highlighting works when ending script block with some other statement or expression**

<img width="583" alt="other-expr" src="https://user-images.githubusercontent.com/92401/229279367-d729977c-9b97-41bd-a1d2-523dc360da90.png">

### Reproduction

This simple snippet is sufficient to to reproduce the bug.

```
<script lang="ts">
export let test: any
</script>
```

### Expected behaviour

I expect the ending `</script>` tag to be correctly highlighted

### System Info

- IDE: VS Code

### Which package is the issue about?

Svelte for VS Code extension

**Comment by jasonlyu123** [comment]
This is an "issue" with the upstream TypeScript syntax highlight. But I can't find a situation where it will break inside a ts file. It was changed to be more precise in multiple lines type alias. But it breaks in vue, svelte and astro files. I am afraid there is much we can do.

**Comment by poppa** [comment]
Ok, so does **upstream** mean VS Code itself? Does this issue get propagated there or should I create a new bug report there (where ever that might be)?

I have never seen this bug inside a pure `.ts` file, so I guess it has something to do with the `<script>` block.

It got worse with the latest VS Code update:

```
Version: 1.77.0
Commit: 7f329fe6c66b0f86ae1574c2911b681ad5a45d63
Date: 2023-03-29T09:57:11.797Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin x64 22.1.0
Sandboxed: Yes
```

Before that only the `type` case was affected, but after this update the `export let varName: Type` problem arose.

**Comment by babak-bxp** [comment]
![1-min](https://user-images.githubusercontent.com/66759640/229598387-8403ced1-b58a-40ce-ae84-62b8ddffc029.png)
![2-min](https://user-images.githubusercontent.com/66759640/229598388-a912c60a-9015-4f3d-8548-0df059b088b5.png)

**Comment by jasonlyu123** [comment]
Type alias without a semi also breaks highlight in a ts file. It's tracked here in #1110. 

About let without a semi, Yes, it is related to the script tag because it doesn't exist in a ts file. So `</script>` won't be a condition for ending the match. Therefore, I don't think they will consider that a bug on their side. 

We have a config file to define regex rules for VSCode to highlight the script tag section with TypeScirpt. But sometimes it seems like if one of the embed language matching rules hasn't ended, It won't consider top-level "</script>". So we either have to change TypeScript's highlight somehow to stop matching type annotation if there is a "</script>" or we'll have to find a way to force any TypeScript matching if there is a "</script>".

We know this problem when it's still in beta. But I haven't found a way to work around it on our side yet. I don't think this will be fixed anytime soon. You can work around it with a comment or an empty line afterwards.

**Comment by JUNGHUNKIM-7** [comment]
same issue here, without semi, syntax crashing happens anywhere.

**Comment by einarpersson** [comment]
Seems to have been fixed here, but I don't know how to update 😆 
(maybe I have to switch to Visual Studio Code Insiders...)

https://github.com/microsoft/TypeScript-TmLanguage/issues/971
https://github.com/microsoft/vscode/issues/179068

**Comment by jasonlyu123** [comment]
Once it's merged into VSCode main, You can test it with VSCode insider. Or you can try it with the TypeScript nightly extension for the nightly release of TSServer and TypeScript-TmLanguage. I tested it, and it seems to be fixed yesterday by another commit. A by-product of other TypeScript statements 😆.

**Comment by einarpersson** [comment]
Is this still an issue for people? I think this is fixed in [vscode 1.77.1](https://github.com/microsoft/vscode/releases/tag/1.77.1). 

If confirmed this could be closed, I suppose?

**Comment by poppa** [comment]
This still seems to be a problem in VSC 1.77.1

<img width="564" alt="Screenshot 2023-04-07 at 07 45 46" src="https://user-images.githubusercontent.com/92401/230549005-c3754a46-c000-4aaf-9577-f43abd6fd70a.png">

**Comment by jasonlyu123** [comment]
It's fixed by another issue other than the linked one, and it's only in the TypeScript nightly extension now.

**Comment by poppa** [comment]
This seems to be fixed in the latest VS Code, so I'm all good 👍 

(And thanks for all the good work you guys do 🤩)

**Comment by jasonlyu123** [comment]
Reopening for visibility. Let's close this when it is fixed in VSCode stable. It should be fixed in a few days in VSCode 1.78.

**Comment by jasonlyu123** [comment]
Fixed in VSCode 1.78
